core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -351,11 +351,11 @@ private[spark] object EventLoggingListener extends Logging {
     // Since we sanitize the app ID to not include periods, it is safe to split on it
     val logName = log.getName.stripSuffix(IN_PROGRESS)
     val codecName: Option[String] = logName.split("\\.").tail.lastOption
-    val codec = codecName.map { c =>
-      codecMap.getOrElseUpdate(c, CompressionCodec.createCodec(new SparkConf, c))
-    }
-
+    
     try {
+      val codec = codecName.map { c =>
```

Comment:
Why would here throw an exception?

Reply:
Because there can throws an exception with "Codec [$codecName] is not available" in CompressionCodec.createCodec function.

Reply:
While we're tightening, this whole body could be 

```
      codecName.map { c =>
        val codec = codecMap.getOrElseUpdate(c, CompressionCodec.createCodec(new SparkConf, c))
        codec.compressedInputStream(in)
      }.getOrElse(in)
```